packages/fuels-types/src/param_types.rs

```diff
@@ -388,6 +385,18 @@ fn try_vector(the_type: &Type) -> Result<Option<ParamType>> {
     Ok(Some(ParamType::Vector(Box::new(vec_elem_type))))
 }
 
+fn try_bytes(the_type: &Type) -> Result<Option<ParamType>> {
+    if !["struct std::bytes::Bytes", "struct Bytes"].contains(&the_type.type_field.as_str()) {
+        return Ok(None);
+    }
+
+    Ok(Some(ParamType::Bytes))
+}
+
+fn try_raw_slice(the_type: &Type) -> Result<Option<ParamType>> {
+    Ok((the_type.type_field == "raw untyped slice").then_some(ParamType::RawSlice))
```

Comment:
Correct me if I'm missing something: if we're only returning an `Ok` here (and not an `Err`), wouldn't it be better if we removed the need for returning `Result<>`? This way, no need to chain two unwraps as we do down there (`try_raw_slice(&the_type).unwrap().unwrap();`). Or making it `Err` if `the_type.type_field == "raw untyped slice"` evals to false. Either way would simplify the use of it (no need for 2 unwraps)

Reply:
Although `try_raw_slice` could drop the `Result`, it has to have the same signature as the other `try_*` `fn`s so that the whole resolving logic might look like this:
```rust
        let matched_param_type = [
            try_primitive,
            try_array,
            try_str,
            try_tuple,
            try_vector,
            try_bytes,
            try_raw_slice,
            try_enum,
            try_struct,
        ]
        .into_iter()
        .map(|fun| fun(the_type))
        .flat_map(|result| result.ok().flatten())
        .next();
```

Some of the other resolving functions can indeed match correctly but fail. For example, `try_array` can determine that the format is indeed that of an array `[_; 10]` but the array has `>1` components which makes it invalid. Saying `None` means _it isn't an array_ which isn't true.

But @hal3e and I have already refactored this and will open up a PR in the next few days. The refactor nuked the whole idea with `try_*` and went with something simpler and more robust.

Feel free to resolve the comment if you're satisfied with the answer.